core/engine.go

```diff
@@ -115,15 +115,21 @@ func (system *System) Start() error {
 
 // Shutdown shuts down all engines in the system.
 func (system *System) Shutdown() error {
-	var err error
-	return system.VisitEnginesE(func(engine Engine) error {
+	var engines []Runnable
+	system.VisitEngines(func(engine Engine) {
 		if m, ok := engine.(Runnable); ok {
-			coreLogger.Infof("Stopping %s...", getEngineName(engine))
-			err = m.Shutdown()
-			coreLogger.Infof("Stopped %s", getEngineName(engine))
+			engines = append(engines, m)
 		}
-		return err
 	})
+	for i := len(engines) - 1; i >= 0; i-- {
+		curr := engines[i]
+		coreLogger.Infof("Stopping %s...", getEngineName(curr))
+		if err := curr.Shutdown(); err != nil {
```

Comment:
should it ignore this error and shutdown the remaining engines?

Reply:
Not sure. There's something to say for either approach.

You could say you try to shut it down again after X time, then force shutdown after a timeout. But that makes it probably more complex than required.

When running in a Docker container, if the system does not shut down before the timeout, it's killed anyways.